### PR TITLE
xray: aws xray tracer span data bug fix

### PR DESCRIPTION
**Pull request by suniltheta**
Commit Message: Fix AWS X-Ray tracer bug where span error was not reported under "Errors and faults" of segment data.
Additional Description: Remove 'error' tag in the trace span from getting reported inside annotations section of xray tracer segment data and instead move it inside "Errors and exceptions" section as per the [xray documentation](https://docs.aws.amazon.com/xray/latest/devguide/xray-api-segmentdocuments.html#api-segmentdocuments-errors).
Risk Level: Low
Testing: Unit tests added
Docs Changes: No
Release Notes: No
Platform Specific Features: No

### CODE REVIEW

File: source/common/quic/platform/quic_logging_impl.h [hunk 1]
```diff
@@ -147,7 +147,7 @@ enum {
 // DFATAL is FATAL in debug mode, ERROR in release mode.
 #ifdef NDEBUG
   LogLevelDFATAL = LogLevelERROR,
-#else  // NDEBUG
```
Comment (suniltheta) [inline source/common/quic/platform/quic_logging_impl.h]: this was unwanted 😬

File: source/extensions/tracers/xray/tracer.cc [hunk 2]
```diff
@@ -203,6 +209,10 @@ void Span::setTag(absl::string_view name, absl::string_view value) {
     // In this case, PeerAddress refers to the client's actual IP address, not
     // the address specified in the HTTP X-Forwarded-For header.
     http_request_annotations_.emplace(SpanXForwardedFor, ValueUtil::boolValue(false));
+  } else if (name == Error) {
+    if (value == True) {
```
Comment (abaptiste) [inline source/extensions/tracers/xray/tracer.cc]: 1.  `if (name == Error && value == True)`
2.  When would we have an Error but the value would be False?

File: source/extensions/tracers/xray/tracer.cc [hunk 3]
```diff
@@ -173,6 +176,8 @@ void Span::setTag(absl::string_view name, absl::string_view value) {
   constexpr auto HttpUserAgent = "user_agent";
```
Comment (abaptiste) [inline source/extensions/tracers/xray/tracer.cc:176]: I know this isn't part of your change here, but perhaps we can look into using the built in headers where this value likely already exists.  If we have to take on a new module dependency here, then ignore me :)

**Review by abaptiste — COMMENTED** [review]
Left a comment.

File: source/extensions/tracers/xray/tracer.cc (hunk 2 above)
Comment (suniltheta) [reply source/extensions/tracers/xray/tracer.cc]: True, we don't have a case where Error tag is sent with false value.

[situation 1](https://github.com/envoyproxy/envoy/blob/main/source/common/tracing/http_tracer_impl.cc#L108)
[situation 2](https://github.com/envoyproxy/envoy/blob/main/source/common/tracing/http_tracer_impl.cc#L255)

I am inclined to getting rid of the second if condition ~`if (value == True) `~ altogether would that be ok?

Comment (suniltheta) [reply source/extensions/tracers/xray/tracer.cc]: Updated this change.
Now doing both name == Error & value == true check in the same line.

**Pull request closed without merging**